File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_100438/typescript/002_word_frequency/solution.ts
import { readFileSync } from "fs";
import { resolve, join } from "path";

const filePath = resolve(join(process.cwd(), "input", "text.txt"));
const content = readFileSync(filePath, "utf-8");

const wordsMap = new Map<string, number>();
const punctuationPattern = /[^\w]/gu;

for (const char of content) {
  if (!char.match(/[a-z]/i)) continue;
}

const text = content.toLowerCase().match(/[\w\s]+/g)?.join("") || "";
const lowercaseText = text.replace(punctuationPattern, "");

const words = lowercaseText.split(/\s+/).filter(word => word.length > 0);

for (const word of words) {
  const cleanedWord = word.match(/[a-z]/gu)?.join("") || "";
  if (cleanedWord && cleanedWord.length > 0) {
    const currentCount = wordsMap.get(cleanedWord) || 0;
    wordsMap.set(cleanedWord, currentCount + 1);
  }
}

const entries = Array.from(wordsMap.entries());
entries.sort((a, b) => {
  if (b[1] !== a[1]) return b[1] - a[1];
  return a[0].localeCompare(b[0]);
});

for (const [word, count] of entries) {
  console.log(`${word}: ${count}`);
}

File: codegen/results/v001/qwen3.5_4b-nvfp4_think/20260409_100438/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"path/filepath"
	"regexp"
	"strings"
)

func main() {
	filePath := "input/text.txt"

	// Read file
	content, err := os.ReadFile(filePath)
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error reading file:", err)
		os.Exit(1)
	}

	text := string(content)

	// Create regex to match words (only letters)
	wordRe := regexp.MustCompile(`[a-z]+`)

	words := wordRe.Split(text, -1)
	for _, w := range words {
		if w != "" {
			text := strings.ToLower(w)
			words = append(words, text)
		}
	}

	// Fix the split - need to match and get the words properly
	var validWords []string
	r := regexp.MustCompile(`[a-zA-Z]+`)
	for _, w := range r.FindAllString(text, -1) {
		validWords = append(validWords, strings.ToLower(w))
	}

	wordCount := make(map[string]int)
	for _, word := range validWords {
		wordCount[word]++
	}

	// Create a slice of struct for sorting
	type WordCount struct {
		Word   string
		Count  int
	}

	var wc []WordCount
	for word, count := range wordCount {
		wc = append(wc, WordCount{word, count})
	}

	// Sort: descending by count, then ascending alphabetically for ties
	sort.Slice(wc, func(i, j int) bool {
		if wc[i].Count != wc[j].Count {
			return wc[i].Count > wc[j].Count
		}
		return wc[i].Word < wc[j].Word
	})

	// Output results
	for _, wc := range wc {
		fmt.Printf("%s: %d\n", wc.Word, wc.Count)
	}
}